src/base/QXmppJingleIq.cpp

```diff
@@ -998,10 +1071,37 @@ void QXmppJingleIq::toXmlElementFromChild(QXmlStreamWriter *writer) const
 
     d->reason.toXml(writer);
 
-    // ringing
-    if (d->ringing) {
-        writer->writeStartElement(QStringLiteral("ringing"));
-        writer->writeDefaultNamespace(ns_jingle_rtp_info);
+    if (d->rtpSessionState) {
+        if (std::holds_alternative<RtpSessionStateActive>(*d->rtpSessionState)) {
+            writer->writeStartElement(QStringLiteral("active"));
+        } else if (std::holds_alternative<RtpSessionStateHold>(*d->rtpSessionState)) {
+            writer->writeStartElement(QStringLiteral("hold"));
+        } else if (std::holds_alternative<RtpSessionStateUnhold>(*d->rtpSessionState)) {
+            writer->writeStartElement(QStringLiteral("unhold"));
+        } else if (auto rtpSessionStateMuting = std::get_if<RtpSessionStateMuting>(&(*d->rtpSessionState))) {
+            if (rtpSessionStateMuting->isMute) {
+                writer->writeStartElement(QStringLiteral("mute"));
+            } else {
+                writer->writeStartElement(QStringLiteral("unmute"));
+            }
+
+            writer->writeDefaultNamespace(ns_jingle_rtp_info);
+
+            if (rtpSessionStateMuting->creator == RtpSessionStateMutingCreator::Initiator) {
+                helperToXmlAddAttribute(writer, QStringLiteral("creator"), QStringLiteral("initiator"));
+            } else if (rtpSessionStateMuting->creator == RtpSessionStateMutingCreator::Responder) {
+                helperToXmlAddAttribute(writer, QStringLiteral("creator"), QStringLiteral("responder"));
+            }
+
+            helperToXmlAddAttribute(writer, QStringLiteral("name"), rtpSessionStateMuting->name);
+        } else {
+            writer->writeStartElement(QStringLiteral("ringing"));
+        }
+
+        if (!std::holds_alternative<RtpSessionStateMuting>(*d->rtpSessionState)) {
+            writer->writeDefaultNamespace(ns_jingle_rtp_info);
+        }
```

Comment:
this is a bit subtle, please just put the `writeDefaultNamespace()` in every case